rpc/gnmi/mixed-schema.md

```diff
@@ -99,6 +99,24 @@ constrained to the specified origin. Particularly:
   the specified `origin`. To delete contents from multiple origins, a client
   MUST specify multiple paths within the `delete` of the `SetRequest`.
 
+### Special considerations for the use of Origin CLI
+* If a `Set` RPC contains `origin` CLI, the CLI origin MUST be given precedence
+  over the OpenConfig origin.  This precedence is independent of the order in
+  which origin CLI appears.  This overrides the gNMI specification requirement
+  of ordered path based precedence within a `SetRequest`.  The goal of this
+  precedence is to create deterministic behavior for resolving overlapping
+  configuration.
+* The `update` operation MUST be supported for `origin` CLI.  The `update`
+  operation is compatible with the requirement for `origin` CLI to take
+  precedence.
+  
+* Because there is no explicit expectation of CLI to use a path based
+  structure, use of the `replace` operation is not recommended unless the
+  intent is to replace the entire configuration with the contents of the origin CLI message.  Replacing
+  the entire configuration with CLI content in a mixed origin `SetRequest`
+  where origin CLI takes precedence means the other origin messages will have
+  no effect.
```

Comment:
Is this correct?

If I send a replace for origin cli, and then openconfig afterwards, does this mean that there is really only CLI even where it does not overlap?

If this is the case I do not think this is intended behaviour - because then there is no way to ever remove something that is already in the CLI without explicitly deleting.

I think the precedence applies only when there is overlap of elements.

Reply:
What I mean here.

If you have the initial state:

```
origin: cli contents: "foobar"
```

and then apply

```
update origin: cli contents: "baz"
update origin: openconfig contents `{"bat": "foo"}`
```

the cli contents "foobar` are never removed, and so at this point one cannot ever do a full configuration replace.

I think you need to rather allow `replace` of the CLI, and say that as long as "bat` does not overlap with "baz" in the CLI, then both are merged.

Reply:
`replace origin=cli` should be treated as a full device config replace.
As such, if `origin=openconfig` is provided it should be specified with a path at root level; otherwise reject it.

I can see this being useful for operators transitioning to use OpenConfig and start moving `cli` towards `openconfig`.

`cli` could also take precedence here if there is a conflict.

Reply:
I have added an update to clarify the replace behavior for mixed OC + CLI.  I appreciate your feedback.